README.md

```diff
@@ -33,309 +33,181 @@ Using RealtimeSanitizer requires RADSan-enabled `clang` and two actions:
+circumstances, like if the mutex is never contended (you might be re-using
+multi-threaded code in a single-threaded context), or if it's only contended at
+times when your system does not require real-time deadlines to be met. RTSan
+pessimistically assumes the worst, which may not be true for your particular
+use case.
 
+You can temporarily disable RTSan inside any scope using an instance of the
+RAII class `__rtsan::ScopedDisabler`:
 
 ```cpp
-__attribute__((no_sanitize("realtime")))
-void mutex_unlock_uncontended (std::mutex& m)
-{
-    m.unlock();
-}
+#include <sanitizer/rtsan_interface.h> // This header is installed with clang
 
-[[clang::realtime]] float process (float x)
+float process (float x) [[clang::nonblocking]]
 {
     ...
-    mutex_unlock_uncontended(m); // I know this is always uncontended, thus real-time safe!
+    {
+        __rtsan::ScopedDisabler _;
+        mutex_.lock(); // I know this is uncontended, thus real-time safe!
+    }
     ...
 }
 ```
 
-## Choice of symbolizer
-
-By default, RADSan uses the installed `llvm-symbolizer` to symbolise the stack
-trace. If you prefer a different symboliser, you can configure RADSan to use it
-by setting the envionment variable `RADSAN_SYMBOLIZER_PATH` at run-time.
-
-
-# How it works
-
-RADSan contains a submodule with a fork of the `llvm-project`. This fork makes
-two additions to the LLVM tool set:
-
-- Additions to the `clang` compiler front end that:
-  - accept `-fsanitize=realtime` as a compiler flag,
-  - accept `[[clang::realtime]]` as a function attribute, and
-  - if RealtimeSanitizer is activated, insert calls to
-    `radsan_realtime_enter()` and `radsan_realtime_exit()` at the entry and
-    exit points of every `[[clang::realtime]]` function. The `radsan` run-time
-    library (described below) is responsible for implementing these methods
-    and performing the sanitization logic.
-
-- A new `radsan` library to LLVM's `compiler-rt` tooling, which:
-  - links to your application at run time,
-  - tracks whether every thread is in a real-time context (i.e. whether the
-    current function or any of its callers in the call stack were marked as
-    `[[clang::realtime]]`), and
-  - intercepts system library calls like `malloc` and `pthread_mutex_lock` at
-    runtime. All interceptors follow the same pattern;
-    - check if the current thread is in a real-time context, and
-    - if not real-time, then call the "real" system library function, or
-    - if real-time, raise an error.
-
-```cpp
-INTERCEPTOR(void *, malloc, SIZE_T size) {
-  radsan::expectNotRealtime("malloc");
-  return REAL(malloc)(size);
-}
-```
-
-# Development
-
-
-## Building the Docker image
-
-Building the Docker image locally is straightforward:
-
-```sh
-make docker
-```
-
-## Running the tests
-
-RADSan follows the existing sanitizer testing patterns, and adds two new test
-targets to the compiler-rt project for each architecture `arch`:
+## Error modes
```

Comment:
Similar, I think we should document the error modes upstream instead of here (one source of truth)

Reply:
Sure. I think maybe I could replace a lot of this simply with a list of "features". Will draft something.

Reply:
I would also include "a full list of features can be found at [[official docs]]"